packages/connectivity/lib/connectivity.dart

```diff
@@ -12,7 +12,28 @@ import 'package:meta/meta.dart';
 /// WiFi: Device connected via Wi-Fi
 /// Mobile: Device connected to cellular network
 /// None: Device not connected to any network
-enum ConnectivityResult { wifi, mobile, none }
+enum ConnectionType { wifi, mobile, none }
+enum ConnectionSubtype { EDGE, HSDPA, LTE, none, unknown }
+
+class ConnectivityResult {
+  ConnectivityResult(this.type, this.subtype);
+
+  final ConnectionType type;
+  final ConnectionSubtype subtype;
+
+  static const ConnectionType wifi = ConnectionType.wifi;
+  static const ConnectionType mobile = ConnectionType.mobile;
+  static const ConnectionType none = ConnectionType.none;
+
+  @override
+  bool operator ==(Object object) {
+    if (!(object is ConnectionType)) return false;
```

Comment:
nits: add {}

Reply:
object here seems will never be a ConnectionType? It is a ConnectivityResult right?

Reply:
So this was the compromise I came to support backward compatibility. But since we are going to include a new API, I think it renders this part moot.